src/pages/workspace/accounting/qbo/export/QuickbooksOutOfPocketExpenseEntitySelectPage.tsx

```diff
@@ -31,33 +31,40 @@ function QuickbooksOutOfPocketExpenseEntitySelectPage({policy}: WithPolicyConnec
     const isLocationsEnabled = Boolean(syncLocations && syncLocations !== CONST.INTEGRATION_ENTITY_MAP_TYPES.NONE);
     const isTaxesEnabled = Boolean(syncTax);
     const policyID = policy?.id ?? '';
+    const isLocationImportEnabled =
+        !policy?.connections?.quickbooksOnline?.config?.syncLocations || policy?.connections?.quickbooksOnline?.config?.syncLocations !== CONST.INTEGRATION_ENTITY_MAP_TYPES.NONE;
 
     const data: CardListItem[] = useMemo(
         () => [
-            {
-                value: CONST.QUICKBOOKS_REIMBURSABLE_ACCOUNT_TYPE.CHECK,
-                text: translate(`workspace.qbo.accounts.check`),
-                keyForList: CONST.QUICKBOOKS_REIMBURSABLE_ACCOUNT_TYPE.CHECK,
-                isSelected: reimbursableExpensesExportDestination === CONST.QUICKBOOKS_REIMBURSABLE_ACCOUNT_TYPE.CHECK,
-                isShown: !isLocationsEnabled,
-            },
             {
                 value: CONST.QUICKBOOKS_REIMBURSABLE_ACCOUNT_TYPE.JOURNAL_ENTRY,
                 text: translate(`workspace.qbo.accounts.journal_entry`),
                 keyForList: CONST.QUICKBOOKS_REIMBURSABLE_ACCOUNT_TYPE.JOURNAL_ENTRY,
                 isSelected: reimbursableExpensesExportDestination === CONST.QUICKBOOKS_REIMBURSABLE_ACCOUNT_TYPE.JOURNAL_ENTRY,
                 isShown: !isTaxesEnabled || isLocationsEnabled,
             },
+        ],
+        [reimbursableExpensesExportDestination, isTaxesEnabled, translate, isLocationsEnabled],
+    );
+
+    if (!isLocationImportEnabled) {
```

Comment:
Forgive my ignorance here but I fail to understand why we had to remove `Vendor bill` and `Check` from within the `data: CardListItem[] = useMemo` and explicitly push it conditionally. Is there a special case that we are handling through `isLocationImportEnabled` instead of `isLocationsEnabled`? Can you please explain a bit on this here?

Reply:
fixed 😄